Source/WebCore/storage/StorageNamespaceProvider.h

```diff
@@ -55,7 +55,7 @@ class StorageNamespaceProvider : public RefCounted<StorageNamespaceProvider> {
     WEBCORE_EXPORT void setSessionIDForTesting(PAL::SessionID);
 
     void setSessionStorageQuota(unsigned quota) { m_sessionStorageQuota = quota; }
-    virtual void copySessionStorageNamespace(Page&, Page&) = 0;
+    virtual void cloneSessionStorageNamespaceForPage(Page&, Page&) { RELEASE_ASSERT_NOT_REACHED(); }
```

Comment:
I don't see a good reason why this can't be pure virtual.

Reply:
After the change, only WebKitLegacy::WebStorageNamespaceProvider should provide the implementation for the function and the function should not be invoked on other inherited class object, so it seems more convenient to add the implementation here.